### PR TITLE
Datasources sidepanel

### LINKED ISSUE #1579: Show streaming sources in left panel in the select sources onboarding step
As a new user
with admin permissions or higher
when i select streaming sources
then i would like to see a side panel with the streaming sources listed in the order they were selected.

they should be added when the streaming source is toggled, and removed when the streaming source is untoggled

### PR DESCRIPTION
**Pull request by ebb-tide**
Closes #1579 

Creates new onboarding sidepanel component. (thanks @alexpaxton!) and uses it to display selected data sources that can be later configured.

### CODE REVIEW

File: ui/src/onboarding/components/side_bar/SideBar.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        <div className="side-bar--container">
+          <h3 className="side-bar--title">{title}</h3>
+          <div className="side-bar--contents">
+            <div className="side-bar--tabs">{this.childTabs}</div>
+            <div className="side-bar--buttons">
+              <ComponentSpacer
+                direction={Direction.Vertical}
+                align={Alignment.Left}
+              >
+                {this.childButtons}
```
Comment (OfTheDelmer) [inline ui/src/onboarding/components/side_bar/SideBar.tsx]: I like the use of childTabs/childButtons getters here

File: ui/src/onboarding/components/side_bar/SideBar.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    const {visible} = this.props
+
+    return classnames('side-bar', {show: visible})
+  }
+
+  private get childTabs(): JSX.Element[] {
+    const {children} = this.props
+    return React.Children.map(children, (child: JSX.Element) => {
+      if (child.type === SideBarTab) {
+        return <SideBarTab {...child.props} />
```
Comment (OfTheDelmer) [inline ui/src/onboarding/components/side_bar/SideBar.tsx]: Could we return `child` here since it's of type `SideBarTab`?

Comment (ebb-tide) [reply ui/src/onboarding/components/side_bar/SideBar.tsx]: we can, yeah.

File: ui/src/onboarding/components/side_bar/SideBar.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
+// Libraries
+import React, {Component} from 'react'
+import classnames from 'classnames'
+
+// Components
+import SideBarTab from 'src/onboarding/components/side_bar/SideBarTab'
+import SideBarButton from 'src/onboarding/components/side_bar/SideBarButton'
+import {ComponentSpacer, Direction, Alignment} from 'src/clockface'
+
+// Styles
```
[the remark this answers is not in the document]
Comment (ebb-tide) [reply ui/src/onboarding/components/side_bar/SideBar.tsx]: @alexpaxton  did it

**Pull request merged**